### ISSUE TITLE
Casing of file in debug infos

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#8868]
I'm not sure if that's a change in VSCode or Haxe, but I now get some weird behavior regarding file casing. For instance in HL I put a breakpoint in `CityView.hx`  and when I break vscode opens a second tab with `cityview.hx`, at the good line, but not showing my breakpoints because they are on "another file".

Actually, I just restarted VSCode and now it works well, so I suppose it's somewhat related to compiler cache ? :-/

**Comment by Simn** [comment]
I also noticed that I had files open with the wrong casing, but I'm not sure if that came from debugging.

There's a chance that this has been a long-standing Haxe issue which used to be hidden by vscode but isn't anymore.

**Comment by ncannasse** [comment]
I think this is somewhat related to compiler cache and/or macros, and yes it somewhat happens on compile time error messages as well.

**Comment by RealyUniqueName** [comment]
Could it be OS-dependent? I've never seen this on Linux.

**Comment by ncannasse** [comment]
Yes, I think we do a to_lower on windows paths (for good reasons) but these "normalized" paths should never end up as module names or position, they're only meant as "unique keys". Which does not seem the case anymore - might be for a long time already and VSCode is suddenly more case-aware than it was before.

**Comment by Simn** [comment]
This just happened to me again in a small project, but restarting the server indeed fixes it and now I cannot reproduce it anymore by following the same steps.

It happened when I moved a typedef from my Main.hx to a standalone file, but forgot to add a necessary import. The standalone file would then error with "Type not found" with a lowercased path.

**Comment by RealyUniqueName** [comment]
I think the reason is here:
https://github.com/HaxeFoundation/haxe/blob/47dd40736e1844af9977ee7ab3e48c8ff0aaf6ad/src/core/path.ml#L98-L104
It's not a safe fix without a stable reproducible sample because of `Path.unique_full_path` being used to set `pfile` of expr positions and current display position. And then `pfile` is used as a key to check various caches or to compare it to other file paths.

**Comment by ncannasse** [comment]
Path.unique_full_path was meant to make sure we don't end up with twice the same file with different casing but that would still point to the same physical file because Windows case insensitive FS.

So it should mostly be used for keys in maps, it should never be used for `pfile` for sure.
PS : this ought to be a relatively recent change, I don't think I had such issues before, although that might be the IDE that was assuming same file for both lowercase and proper-cased paths

**Comment by nadako** [comment]
I found one reproducable case for this one:
```haxe
vfs.putContent("I.hx", "interface I {}");
vfs.putContent("C.hx", "class C implements I { public function new() {} }");
vfs.putContent("Main.hx", "class Main { static function main() f(new C()); static function f(i:I) {} }");

var args = ["-main", "Main", "-js", "out.js"];

runHaxe(args);
vfs.putContent("I.hx", "interface I<T> {}");
runHaxeJson([], ServerMethods.Invalidate, {file: new FsPath("C.hx")});
runHaxeJson(args, DisplayMethods.Completion, {
	file: new FsPath("Main.hx"),
	contents: "class Main { static function main() f(new C()); static function f(i:I<Int) {} }",
	offset: 73,
	wasAutoTriggered: false
});
vfs.putContent("Main.hx", "class Main { static function main() f(new C()); static function f(i:I<Int>) {} }");
runHaxe(args);

trace(lastResult.stderr);
```

not sure if it can be reduced further, but this will trace unique file path

**Comment by ncannasse** [comment]
It's still an issue, and a very annoying one because it also affects the Problems window (and terminal), leading to having two opened files which are the same with different casing, and that leads to numerous issues with changes propagation.

And debugging experience too because of the way breakpoints are handled in HL or JS, you cannot set breakpoints because they will be set on an unrecognized file (ProperCase.hx instead propercase.hx).

Should be fixed urgently please :'(

**Comment by RealyUniqueName** [comment]
Can't we just say "Haxe compiler is case sensitive" and remove lowercasing from `unique_full_path`?
So it's up to users to keep hx files properly named.

**Comment by Simn** [comment]
This can probably be closed after merging #9283.
